account_analytic_parent/models/account_analytic_account.py

```diff
@@ -59,3 +59,14 @@ def name_get(self):
                 current = current.parent_id
             res.append((account.id, name))
         return res
+
+    @api.multi
+    def archive_account(self):
+        for account in self:
+            account.active = False
+
+    @api.multi
+    def write(self, vals):
+        if 'active' in vals and not vals['active']:
+            self.mapped('child_ids').archive_account()
+        return super(AccountAnalyticAccount, self).write(vals)
```

Comment:
perhaps you prefer this

```python
@api.multi
    def archive_account(self, unarchive=False):
        for account in self:
            account.active = unarchive

    @api.multi
    def write(self, vals):
        if 'active' in vals and not vals['active']:
            self.mapped('child_ids').archive_account()
        res = super(AccountAnalyticAccount, self).write(vals)
        if 'active' in vals and vals['active'] and self.env.context('unarchive_children', False):
            self.mapped('child_ids').archive_account(True)
        return res
```

This way the unarchiving of all children is also possible, but you shall pass a context to do that 😏😉
(the unarchiving of children is done after the unarchive of the parent to make this compatible with the contrain method I put you before)

Reply:
It's ok, better to leave it like this. I can overwrite this in custom module.